code/modules/mob/living/carbon/life.dm

```diff
@@ -460,6 +460,9 @@
 	for(var/obj/item/bodypart/limb as anything in bodyparts)
 		. |= limb.on_life(seconds_per_tick, times_fired)
 
+	for(var/datum/bodypart_effect/effect as anything in bodypart_effects)
+		effect.on_life(src, seconds_per_tick, times_fired)
```

Comment:
Rather than adding this to `life` directly, could we have the datum register the life signal?  
Not sure if it's _better_ to do that way (you would still need the list of effects on `/carbon` for tracking reapplication, unless you move to using the status effect interface), but it might be neater to have the behaviour contained that way.

Reply:
Sure, I don't feel strongly about this either way